src/mini_trainer/osft_utils.py

```diff
@@ -10,6 +10,9 @@
 
 from mini_trainer.utils import log_rank_0, check_distributed_is_synchronized
 
+# Memory optimization constants
+OSFT_CACHE_CLEAR_INTERVAL = 5  # Clear GPU cache every N parameters during matrix reconstruction
```

Comment:
Would it make sense to make this configurable?

Reply:
I don't think this needs to be configurable. This is an internal memory optimization detail during checkpoint saving similar to how we wouldn't expose garbage collection frequency or other memory management params. Value of 5 works but making it configurable would add unnecessary API complexity.

Reply:
@NikhilNayak-debug It could be worthwhile to have it read this value from an env like:
```suggestion
OSFT_CACHE_CLEAR_INTERVAL = os.getenv('OSFT_CACHE_CLEAR_INTERVAL', 5)
```

So that it still uses the default but is something we can easily adjust if needed.

Reply:
Sounds good made the change!